raw/src/requests/_base/errors.rs

```diff
@@ -1,16 +1,57 @@
+use std::error;
+use std::fmt;
+
 use crate::types::*;
 
-error_chain! {
-    foreign_links {
-        Json(::serde_json::Error);
+#[derive(Debug)]
+pub struct Error(ErrorKind);
+
+#[derive(Debug)]
+pub(crate) enum ErrorKind {
+    EmptyBody,
+    TelegramError {
+        description: String,
+        parameters: Option<ResponseParameters>,
+    },
+    DetachedError(String),
+    Json(::serde_json::Error),
+}
+
+impl From<::serde_json::Error> for ErrorKind {
```

Comment:
Same as the previous comment:
Wouldn't it be better to make it `impl From<> for Error` directly? To reduce boilerplate

Reply:
I thought about this. It's implemented for `ErrorKind` instead of `Error` because `From` implementation it's a part of public interface. User can use this implementation (and I don't think it's intended) and we can't just remove any of them without breaking backward compatibility. 

```rust
use telegram_bot::{Api, Error};

#[tokio::main]
fn main() -> Result<(), Error> {
    let token = String::from_utf8_lossy(std::fs::read(".token")?);
    let api = Api::new(token);
    ...
}
```

For example, we have `From<std::io::Error>` implementation, as far as I remember we only have it because `tokio::fs::read` can return IoError, but we can decide, we want another way to read files asynchronously, and may be this way will use it's own error type instead of `std::io::Error` and we'll have to remain this implementation.

Of course it make sense only if `ErrorKind` is private. What do you think if we leave it private but implement `Error::source` for it, so user can check it?